segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageProvider.java

```diff
@@ -0,0 +1,179 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.chunklayer;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Defines an abstraction for Permanent Storage.
+ * Note: not all operations defined here are needed.
+ */
+public interface ChunkStorageProvider extends AutoCloseable {
+    /**
+     * Gets a value indicating whether this Storage implementation supports truncate operation on underlying storage object.
+     *
+     * @return True or false.
+     */
+    boolean supportsTruncation();
+
+
+    /**
+     * Gets a value indicating whether this Storage implementation supports append operation on underlying storage object.
+     *
+     * @return True or false.
+     */
+    boolean supportsAppend();
+
+
+    /**
+     * Gets a value indicating whether this Storage implementation supports merge operation either natively or through appends.
+     *
+     * @return True or false.
+     */
+    boolean supportsConcat();
+
+    /**
+     * Gets a value indicating whether this Storage implementation supports native merge operation on underlying storage object.
+     *
+     * @return True or false.
+     */
+    boolean supportsNativeConcat();
+
+    /**
+     * Determines whether named file/object exists in underlying storage.
+     *
+     * @param chunkName Name of the storage object to check.
+     * @return True if the object exists, false otherwise.
+     * @throws IOException Throws IOException in case of I/O related exceptions.
+     */
+    boolean exists(String chunkName) throws IOException;
+
+    /**
+     * Creates a new file.
+     *
+     * @param chunkName String name of the storage object to create.
+     * @return ChunkHandle A writable handle for the recently created chunk.
+     * @throws IOException Throws IOException in case of I/O related exceptions.
+     */
+    ChunkHandle create(String chunkName) throws IOException;
```

Comment:
what about having an async API ?
  CompletableFuture<ChunkHandle> create(String chunkName)

the same at least for delete,read,write, openRead and openRead, concat....
if you have a fully async storage you will be able to leverage its features.
but if we start with a blocking interface we will never  be able to leverage async features of backing storage

Reply:
That was the original plan. However, the required changes were getting bit complicated. I wasn't sure this added complexity was actually buying a lot of performance. (Java threads doing IO do get blocked anyway. ) So I decided to do it in small steps. In this iteration still keeping the sync interface. In next iteration or two I do plan to turn this into async operation.